tests/error/test_array_errors.py

```diff
@@ -0,0 +1,20 @@
+import pathlib
+import pytest
+
+from tests.error.util import run_error_test
+
+path = pathlib.Path(__file__).parent.resolve() / "array_errors"
+files = [
+    x
+    for x in path.iterdir()
+    if x.is_file()
+    if x.suffix == ".py" and x.name != "__init__.py"
```

Comment:
This is consistent with all the other error tests, but shouldn't the double if be an `and` chain instead?

Reply:
Yes, that would be better. I'll make a PR to change it across the board